src/solver/optim-model-filler/LinearExpression.cpp

```diff
@@ -117,4 +125,115 @@ LinearExpression LinearExpression::negate() const
 {
     return {-offset_, scale_map(coef_per_var_, -1)};
 }
+
+TimeDependentLinearExpression::TimeDependentLinearExpression(const std::vector<unsigned>& timesteps)
+{
+    for (auto timestep: timesteps)
+    {
+        linearExpressions_[timestep] = LinearExpression();
+    }
+}
+
+TimeDependentLinearExpression::TimeDependentLinearExpression(
+  const std::vector<unsigned>& timesteps,
+  const LinearExpression& linearExpression)
+{
+    for (auto timestep: timesteps)
+    {
+        linearExpressions_[timestep] = linearExpression;
+    }
+}
+
+TimeDependentLinearExpression::TimeDependentLinearExpression(
+  const std::map<unsigned int, LinearExpression>& linearExpressions):
+    linearExpressions_(linearExpressions)
+
+{
+}
+
+TimeDependentLinearExpression TimeDependentLinearExpression::operator+(
+  const TimeDependentLinearExpression& other) const
+{
+    const auto& linear_expressions = GetLinearExpressions();
+    const auto& other_linear_expressions = other.GetLinearExpressions();
+
+    std::map result(linear_expressions);
+    for (auto [key, value]: other_linear_expressions)
+    {
+        if (result.contains(key))
+        {
+            result[key] += value;
+        }
+        else
+        {
+            result[key] = value;
+        }
+    }
+
+    return TimeDependentLinearExpression(std::move(result));
+}
+
+TimeDependentLinearExpression TimeDependentLinearExpression::operator-(
+  const TimeDependentLinearExpression& other) const
+{
+    // checkOtherLength(other);
+    const auto& linear_expressions = GetLinearExpressions();
+    const auto& other_linear_expressions = other.GetLinearExpressions();
+    std::map<unsigned int, LinearExpression> result;
+    for (size_t i = 0; i < linear_expressions.size(); ++i)
+    {
+        result[i] = linear_expressions.at(i) - other_linear_expressions.at(i);
```

Comment:
When adding or subtracting linear expression, we may end up with variables having coefficient zero (as they are not deleted from the map), does this lead to errors when adding such variables in the solver ? My insight is that it could...